### PR TITLE
Data provider apteka-911

### PR DESCRIPTION
**Pull request by ritalinch**

### CODE REVIEW

**Comment by MuchosChuchos** [comment]
1) So many files are changed that do not belong to your data source.
2) You have created interface HTMLParser and class SocialnaParser in the wrong package (dto??? Does parser belongs to dtos??)
If I have understood you correctly you create HTMLParser for SocialnaParser, but any class could have that interface. 

this is what catches the eye the first time.

**Comment by MuchosChuchos** [comment]
I cannot understand the commits history at all

**Review by VolodymyrHoliev — APPROVED** [review]

File: api/src/main/resources/application.yaml [hunk 1]
```diff
@@ -50,3 +50,7 @@ pharmagator:
       medicament-category-id: !!str 4628712
       pharmacy-name: pharmacy-rozetka
       page-limit: 5
+    apteka-socialna:
+      url: https://1sa.com.ua/medikamenty
+      pharmacy-name: apteka-911-socialna
+      page-limit: 10
```
Comment (Oleksandr-Eleks) [inline api/src/main/resources/application.yaml:56]: Please limit to 5

File: api/src/test/java/com/eleks/academy/pharmagator/dataproviders/dto/socialna/parsers/SocialnaParserTestIT.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        resultDataSet.forEach(Assertions::assertNotNull);
+
+        resultDataSet.forEach(medicineDto -> assertEquals(PHARMACY_NAME, medicineDto.getPharmacyName()));
+    }
+
+    @Test
+    void getMedicinesFromPageByUrl_malformedUrl_IAE(){
+        assertThrows(IllegalArgumentException.class, ()-> subject.getMedicinesFromPageByUrl("qwerty"));
+    }
+}
```
Comment (Oleksandr-Eleks) [inline api/src/test/java/com/eleks/academy/pharmagator/dataproviders/dto/socialna/parsers/SocialnaParserTestIT.java:77]: Please add BL in the end

File: api/src/test/resources/html/socialna-test-page.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
```
Comment (Oleksandr-Eleks) [inline api/src/test/resources/html/socialna-test-page.html:1]: Please remove this file from this PR

**Review by Oleksandr-Eleks — CHANGES_REQUESTED** [review]

File: api/src/test/resources/html/socialna-test-page.html (hunk 3 above)
Comment (VolodymyrHoliev) [reply api/src/test/resources/html/socialna-test-page.html:1]: That's for testing. This provider is parsing html pages. I thought that it's not the best idea to hardcode html page as a string in test classes. Am I wrong and it's better to use plain strings instead?

Comment (Oleksandr-Eleks) [reply api/src/test/resources/html/socialna-test-page.html:1]: My bad. 
This is good idea.

File: api/src/test/java/com/eleks/academy/pharmagator/dataproviders/dto/socialna/parsers/SocialnaParserTestIT.java (hunk 2 above)
Comment (VolodymyrHoliev) [reply api/src/test/java/com/eleks/academy/pharmagator/dataproviders/dto/socialna/parsers/SocialnaParserTestIT.java:77]: I have fixed this issues and made PR on Rita's branch. She'll update this PR a little bit later.

File: api/src/main/resources/application.yaml (hunk 1 above)
Comment (VolodymyrHoliev) [reply api/src/main/resources/application.yaml]: Fixed

File: api/src/test/java/com/eleks/academy/pharmagator/dataproviders/dto/socialna/parsers/SocialnaParserTestIT.java (hunk 2 above)
Comment (VolodymyrHoliev) [reply api/src/test/java/com/eleks/academy/pharmagator/dataproviders/dto/socialna/parsers/SocialnaParserTestIT.java]: Fixed

**Review by Oleksandr-Eleks — APPROVED** [review]

**Review by Anna-Kudiakova — APPROVED** [review]

**Pull request merged**